perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -856,16 +858,31 @@ public void removeAdmin(PerunSession sess, Group group, Group authorizedGroup) t
 	}
 
 	@Override
-	public Paginated<RichGroup> getGroupsPage(PerunSession sess, Vo vo, GroupsPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException {
+	public Paginated<RichGroup> getGroupsPage(PerunSession sess, Vo vo, GroupsPageQuery query, List<String> attrNames) throws VoNotExistsException, PrivilegeException, MemberNotExistsException, GroupNotExistsException, MemberGroupMismatchException {
 		Utils.checkPerunSession(sess);
 		perunBl.getVosManagerBl().checkVoExists(sess, vo);
 
-		if (!AuthzResolver.authorizedInternal(sess, "getGroupsPage_Vo_GroupsPageQuery_List<String>_policy", vo)) {
+		Member member = query.getMemberId() == null ? null : perunBl.getMembersManagerBl().getMemberById(sess, query.getMemberId());
+		if (member != null) {
+			perunBl.getMembersManagerBl().checkMemberExists(sess, member);
+		}
+
+		// Authorization
+		if (member != null) {
+			if (!AuthzResolver.authorizedInternal(sess, "member-getGroupsPage_Vo_GroupsPageQuery_List<String>_policy",  member)) {
+				throw new PrivilegeException(sess, "getGroupsPage");
+			}
+		} else if (!AuthzResolver.authorizedInternal(sess, "getGroupsPage_Vo_GroupsPageQuery_List<String>_policy", vo)) {
 			throw new PrivilegeException(sess, "getGroupsPage");
 		}
 
 		Paginated<RichGroup> result = groupsManagerBl.getGroupsPage(sess, vo, query, attrNames);
-		result.setData(getGroupsManagerBl().filterOnlyAllowedAttributes(sess, result.getData(), null, true));
+
+		if (member != null) {
+			result.setData(getGroupsManagerBl().filterOnlyAllowedAttributes(sess, result.getData(), member, null, true));
+		} else {
```

Comment:
This can be even more simplified, there is no need for IF statement - the IF block should be sufficient for both cases, because member parameter can be null in method `filterOnlyAllowedAttributes`.